src/java/org/apache/cassandra/net/MessageOut.java

```diff
@@ -96,6 +97,11 @@
     //the second object is the POJO to serialize
     public final List<Object> parameters;
 
+    /**
+     * Allows sender to explicitly state which connection type the message should be sent on.
+     */
+    public final ConnectionType connectionType;
```

Comment:
A part of me wants to say don't make MessageOut bigger, but it's probably a drop in the bucket in terms of allocation rate for processing network messages.